diff --git sapphire/core/model/DB.php sapphire/core/model/DB.php
index 6a25841..c942a99 100755
--- sapphire/core/model/DB.php
+++ sapphire/core/model/DB.php
@@ -32,6 +32,16 @@ class DB {
 	 * Internal flag to keep track of when db connection was attempted.
 	 */
 	private static $connection_attempted = false;
+	
+	protected static $profiler = null;
+
+	public static function set_profiler($profiler) {
+		self::$profiler = $profiler;
+	}
+
+	public static function remove_profiler() {
+		self::$profiler = null;
+	}
 
 	/**
 	 * Set the global database connection.
@@ -105,6 +115,13 @@ class DB {
 	static function query($sql, $errorLevel = E_USER_ERROR) {
 		self::$lastQuery = $sql;
 		
+		if(isset(self::$profiler)) {
+			self::$profiler->start($sql);
+			$result = self::getConn()->query($sql, $errorLevel);
+			self::$profiler->stop();
+			return $result;
+		}
+		
 		return self::getConn()->query($sql, $errorLevel);
 	}
 
